src/js/media-controller.js

```diff
@@ -1007,66 +1016,37 @@ const Delegates = {
   },
 
   [MediaUIAttributes.MEDIA_STREAM_TYPE](el) {
-    const media = el.media;
-
-    if (!media) return null;
-
-    const duration = media.duration;
-
-    if (duration === Infinity) {
-      return StreamTypes.LIVE;
-    } else if (Number.isFinite(duration)) {
-      return StreamTypes.ON_DEMAND;
-    } else {
-      const defaultType = el.getAttribute('default-stream-type');
-
-      if (StreamTypeValues.includes(defaultType)) {
-        return defaultType;
-      }
-    }
-
-    return null;
+    return getStreamType(el);
+  },
+  [MediaUIAttributes.MEDIA_TARGET_LIVE_WINDOW](el) {
+    return getTargetLiveWindow(el);
   },
   [MediaUIAttributes.MEDIA_TIME_IS_LIVE](controller) {
     const media = controller.media;
     
     if (!media) return false;
-
-    const streamIsLive = controller.getAttribute(MediaUIAttributes.MEDIA_STREAM_TYPE) === 'live';
-    const seekable = media.seekable;
-
-    // If there's no way to seek, assume the media element is keeping it "live"
-    if (streamIsLive && !seekable) {
-      return true;
+    if (typeof media.liveEdgeStart === 'number') {
+      if (Number.isNaN(media.liveEdgeStart)) return false;
+      return media.currentTime >= media.liveEdgeStart;
     }
 
-    // If the slotted media does not provide a `seekable` property or its length is falsey,
-    // assume the media is not live.
-    if (!seekable?.length) {
-      return false;
-    }
+    const live = getStreamType(controller) === 'live';
+    // Can't be playing live if it's not a live stream
+    if (!live) return false;
+    const seekable = media.seekable;
+    // If the slotted media element is live but does not expose a 'seekable' `TimeRanges` object,
+    // always assume playing live
+    if (!seekable) return true;
+    // If there is an empty `seekable`, assume we are not playing live
+    if (!seekable.length) return false;
 
     // Default to 10 seconds
     // Assuming seekable range already accounts for appropriate buffer room
-    let liveThreshold = 10;
-    let liveThresholdAttr = controller.getAttribute('livethreshold');
-
-    if (liveThresholdAttr !== null) {
-      liveThresholdAttr = Number(liveThresholdAttr);
-
-      if (!Number.isNaN(liveThresholdAttr)) {
-        liveThreshold = liveThresholdAttr;
-      }
-    }
-
-    const currentTime = media.currentTime;
-    const seekableEnd = seekable.end(seekable.length - 1);
-
-    if (currentTime > seekableEnd - liveThreshold) {
-      return true;
-    }
-
-    return false;
+    const liveEdgeStartOffset = controller.hasAttribute('livethreshold') 
```

Comment:
Should this attribute be with respect to the new proposal? Maybe to match the variable name?

Reply:
Yeah I agree this attribute should get a rename. Since the proposal moved to `liveEdgeStart`, there isn't an explicit equivalent of `livethreshold` (it's effectively the variable name `liveEdgeStartOffset`). Do you think the attribute should be `liveedgestartoffset`? or `liveedgeoffset` (maybe a bit confusing, though that was the name of the property in the previous iteration of the proposal)? or something else? Def open to suggestions here (Names Are Hard™️)

Reply:
`liveedgeease` :) 

`liveedgeoffset` sounds pretty good to me

Reply:
Cool I'll run with that unless folks have any concerns.